src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarm.java

```diff
@@ -487,8 +490,45 @@ public ListSuitesResult listSuites(String runArn) {
 
         return api.listSuites(request);
     }
-
-
+    
+    /*
+     * Check if account has unmetered devices based on upload type
+     */
+    public void checkAccountUnmeteredStatus(String appArtifact) throws AWSDeviceFarmException {
```

Comment:
You are correct.
Fixed it. We get the unmetered count and then throw exception in awsdevicefarmrecorder class if unmetered device count is less than 0